metacat-connector-hive/src/main/java/com/netflix/metacat/connector/hive/sql/DirectSqlGetPartition.java

```diff
@@ -459,7 +465,20 @@ public int getPartitionCount(
                         PartitionInfo.builder().name(QualifiedName.ofPartition(catalogName,
                             databaseName, tableName, name)).auditInfo(auditInfo).serde(storageInfo).build()));
                 }
+
+                // Fail if the number of partitions exceeds the threshold limit.
+                if (result.size() > config.getMaxPartitionsThreshold()) {
+                    registry.counter(registry.createId(HiveMetrics.CounterHiveGetPartitionsExceedThresholdFailure
+                            .getMetricName()).withTags(tableQName.parts())).increment();
+                    final String message =
+                        String.format("Number of partitions queried for table %s exceeded the threshold %d",
+                            tableQName, config.getMaxPartitionsThreshold());
+                    log.warn(message);
+                    throw new IllegalArgumentException(message);
+                }
             }
+            registry.gauge(registry.createId(HiveMetrics.GaugePreExpressionFilterGetPartitionsCount
+                .getMetricName()).withTags(tableQName.parts())).set(noOfRows);
             return result;
```

Comment:
What's the purpose of this metrics? Do we need to know the number of getPartitions before applying the filter? To report the failure case, the above threshhold metric would be useful.

Reply:
Yes, this metric will be helpful to monitor the direct sql. This will tell us if we are filtering the partitions on the client side or on the server side.

Reply:
ok.